sherpa/optmethods/optfcts.py

```diff
@@ -780,7 +781,10 @@ def stat_cb1( pars ):
     #     rv = (status, x, fval, msg, {'info': info, 'nfev': nfev,
     #                                  'njev': njev, 'covar': covar})
     # else:
-    rv = (status, x, fval, msg, {'info': info, 'nfev': nfev,
-                                 'covar': covar})
+    if info != 1:
```

Comment:
The logic here is hard to follow, given how info is changed, but can we guarantee at this point that `covar` is defined? I think so, but this depends on `cpp_lmdif` only ever returning a value >= 0